app/views/govuk_publishing_components/components/_date_input.html.erb

```diff
@@ -0,0 +1,85 @@
+<%
+  id ||= "input-#{SecureRandom.hex(4)}"
+  name ||= nil
+  items ||= nil
+
+  legend_text ||= nil
+  hint ||= nil
+  error_message ||= nil
+  describedby ||= nil
+
+  has_error = error_message
+
+  css_classes = %w(gem-c-date-input govuk-date-input)
+  form_group_css_classes = %w(govuk-form-group)
+  form_group_css_classes << "govuk-form-group--error" if has_error
+
+  hint_id = "hint-#{SecureRandom.hex(4)}"
+  error_id = "error-#{SecureRandom.hex(4)}"
+
+  aria_described_by ||= nil
+  if hint || has_error || describedby
+    aria_described_by = []
+    aria_described_by << hint_id if hint
+    aria_described_by << error_id if has_error
+    aria_described_by << describedby if describedby
+    aria_described_by = aria_described_by.join(" ")
+  end
+
+%>
+
+<% date_input_items = items ? items : date_input_items = [
```

Comment:
This is tricky to understand, as contains two assignments. Could this just be the default for `items`?